s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/RecordProcessor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.s3.source.utils;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.storage.Converter;
+
+import io.aiven.kafka.connect.s3.source.config.S3SourceConfig;
+import io.aiven.kafka.connect.s3.source.input.Transformer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class RecordProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecordProcessor.class);
+
+    private RecordProcessor() {
+
+    }
+
+    public static List<SourceRecord> processRecords(final Iterator<AivenS3SourceRecord> sourceRecordIterator,
+            final List<SourceRecord> results, final S3SourceConfig s3SourceConfig,
+            final Optional<Converter> keyConverter, final Converter valueConverter,
+            final AtomicBoolean connectorStopped, final Transformer transformer, final FileReader fileReader,
+            final OffsetManager offsetManager) {
+
+        final Map<String, String> conversionConfig = new HashMap<>();
+        final int maxPollRecords = s3SourceConfig.getInt(S3SourceConfig.MAX_POLL_RECORDS);
+
+        for (int i = 0; sourceRecordIterator.hasNext() && i < maxPollRecords && !connectorStopped.get(); i++) {
```

Comment:
The S3 API should handle the maxrecords for us, so we should remove that hear, as even if there is a bug, we need to handle every record we get out of the S3 poll, as when we go to next() token we will get a new batch of records and miss some records if we don't just process everything S3 has given us.

Reply:
May not be true always. Users can configure max records for poll, and batch config in s3 api. 
Both these configs have their own unique functionality. Poll method takes care of it depending on it's thread status.

Reply:
I was referring to batch config in s3 api

Reply:
Sorry Murali, I've been trying to understand the config for batch but I can't find a reference to it in the S3 API, can you add a link please?

Reply:
I was thinking of api withMaxKeys which limits per fetch